README.rst

```diff
@@ -84,12 +86,27 @@ Example settings
 Advanced customisation
 ----------------------
 
-Getting IP
-~~~~~~~~~~
+Getting IP Address
+~~~~~~~~~~~~~~~~~~
+
+By default, ``BasicAuthIPWhitelistMiddleware`` uses ``request.META["REMOTE_ADDR"]``
+as the client's IP, which corresponds to the IP address connecting to Django.
+If you have a reverse proxy (eg ``nginx`` in front), this will result in the IP address of
+``nginx``, not the client.
+
+Correctly determining the IP address can vary between deployments. Guessing incorrectly can
+result in security issues. Instead, this library requires you configure this yourselves.
```

Comment:
**Thoughts:** We should completely remove the `BASIC_AUTH_WHITELISTED_IP_NETWORKS` setting, and the "get client ip" function. We should probably call it just a `BASIC_AUTH_EXTRA_CHECK_FUNCTION` or something like that, and make it return True if the request passes the check. That way this package stops doing the IP checks explicitly altogether.

We should add a recipe in the README for adding an IP check though.